### PR TITLE
Add fake impl for aten._unique

### LINKED ISSUE #160664: [aot dispatch] Support mutations on tensors created in-graph
### 🐛 Describe the bug

I'm hitting this assertion:

https://github.com/pytorch/pytorch/blob/65053c03a3d209060cb239d20a229dac37cf9dd1/torch/_functorch/_aot_autograd/functional_utils.py#L447-L457

This happens in a few inductor tests when I enable compiled autograd on them e.g. `PYTORCH_TEST_WITH_INDUCTOR=1 pytest test/test_ops.py::TestCommonCPU::test_noncontiguous_samples_index_fill_cpu_float3`. One affected autograd op is IndexFillBackward1, whose impl can create new tensors and call copy_range on them:

```cpp
static variable_list IndexFillBackward1_apply_functional(
  variable_list&& grads,
  std::array<bool,2> needs_input_grad, int64_t& dim, Tensor& index)
{
  IndexRangeGenerator gen;
  auto self_ix = gen.range(1);
  auto value_ix = gen.range(1);
  variable_list grad_inputs(gen.size());
  const auto& grad = grads[0];
  bool any_grad_defined = any_variable_defined(grads);
  if (needs_input_grad[/*self*/0]) {
    auto grad_result = any_grad_defined ? (grad.index_fill(dim, index, 0)) : Tensor();
    copy_range(grad_inputs, self_ix, grad_result);
  }
  if (needs_input_grad[/*value*/1]) {
    auto grad_result = any_grad_defined ? (grad.index_select(dim, std::get<0>(at::_unique(index, /*sorted=*/false))).sum()) : Tensor();
    copy_range(grad_inputs, value_ix, grad_result);
  }
  return grad_inputs;
}
inline variable_list IndexFillBackward1_apply_functional_ivalue(const variable_list& grads, const ivalue_list& args)
{
#ifdef C10_MOBILE
  TORCH_INTERNAL_ASSERT(false, "compiled autograd doesn't work on mobile");
#else
  auto packed_args = PackedArgs(args);
  auto needs_input_grad = packed_args.unpack<std::array<bool, 2>>();
  auto dim = packed_args.unpack<int64_t>();
  auto index = packed_args.unpack<Tensor>();
  return IndexFillBackward1_apply_functional(variable_list(grads), needs_input_grad, dim, index);
#endif
}
```

When we trace a graph with IndexFillBackward1 in it, we don't seem to be able to functionalize the copy_range, and `copy_` ops stay in the aot graphs

### Versions

main, patching compiled autograd on all inductor wrapped tests

### PR DESCRIPTION
**Pull request by jansel**
Stack from [ghstack](https://github.com/ezyang/ghstack/tree/0.15.0) (oldest at bottom):
* __->__ #185318

IndexFillBackward1 uses aten._unique.default on the index tensor when computing the gradient for the fill value. Under compiled autograd, FakeTensor did not have a dedicated implementation for aten._unique.default, and the operator was not tagged as dynamic-output-shape. That allowed tracing/Inductor to see a bogus fixed unique length and compile the value-gradient path against the wrong shape, which could silently produce an incorrect scalar gradient when size assertions were disabled in the inductor-wrapped OpInfo tests.

Add a FakeTensor implementation for aten._unique.default that shares the existing dynamic unique shape logic with aten._unique2.default, and tag the schema as dynamic_output_shape for generated operator metadata. The regression test enables dynamic-output-shape capture and verifies compiled autograd compiles through the unique path without graph-breaking, preserving the correct value gradient for noncontiguous index_fill inputs.

Fixes #160664

Generated by my agent

Test Plan:

python -m pytest -q test/inductor/test_compiled_autograd.py::TestCompiledAutograd::test_index_fill_value_grad_noncontiguous

python -m pytest -q test/test_fake_tensor.py::FakeTensorTest::test_unique_output_dtype test/test_fake_tensor.py::FakeTensorTest::test_unique_default_dynamic_output_shape

PYTORCH_TEST_WITH_INDUCTOR=1 python - <<'PY'

import pytest

import torch._dynamo.config

torch._dynamo.config.compiled_autograd = True

raise SystemExit(pytest.main([

    '-q',

    'test/test_ops.py::TestCommonCPU::test_noncontiguous_samples_index_fill_cpu_float32',

]))

PY

lintrunner -a

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

### CODE REVIEW

**Comment by jansel** [comment]
@claude review these changes

**Review by eellison — APPROVED** [review]

**Comment by jansel** [comment]
My agent says I rebased the PR onto current main and fixed the CI failures that were caused by stale index_fill expected-failure metadata. The AOT symbolic and fake-crossref backward index_fill tests now pass locally, and the PR keeps the aten._unique fake-tensor/dynamic-shape fix.

**Comment by jansel** [comment]
@claude are the CI failures caused by this PR?
